karapace/kafka/consumer.py

```diff
@@ -66,3 +69,209 @@ def get_watermark_offsets(
             return result
         except KafkaException as exc:
             raise_from_kafkaexception(exc)
+
+    def commit(
+        self,
+        message: Message | None = None,
+        offsets: list[TopicPartition] | None = None,
+        _asynchronous: bool = False,
+    ) -> list[TopicPartition] | None:
+        """Commit offsets based on a message or offsets (topic partitions).
+
+        The `message` and `offsets` parameters are mutually exclusive.
+        """
+        if message is not None and offsets is not None:
+            raise ValueError("Parameters message and offsets are mutually exclusive.")
+
+        try:
+            if message is not None:
+                return super().commit(message=message, asynchronous=False)
+
+            return super().commit(offsets=offsets, asynchronous=False)
+        except KafkaException as exc:
+            raise_from_kafkaexception(exc)
+
+    def committed(self, partitions: list[TopicPartition], timeout: float | None = None) -> list[TopicPartition]:
+        try:
+            if timeout is not None:
+                return super().committed(partitions, timeout)
+
+            return super().committed(partitions)
+        except KafkaException as exc:
+            raise_from_kafkaexception(exc)
+
+    def subscribe(  # type: ignore[override]
+        self,
+        topics: list[str] | None = None,
+        patterns: list[str] | None = None,
+    ) -> None:
+        """Subscribe to a list of topics and/or topic patterns.
+
+        Subscriptions are not incremental.
+        For `Consumer.subscribe`, Topic patterns must start with "^", eg.
+        "^this-is-a-regex-[0-9]", thus we prefix all strings in the `patterns`
+        list with "^".
+
+        The `on_assign` and `on_revoke` callbacks are set to keep track of
+        subscriptions (topics).
+
+        More in the confluent-kafka documentation:
+        https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Consumer.subscribe
+        """
+        topics = topics or []
```

Comment:
```assert self._subscription == frozenset(), "expected to call `unsubscribe` before attempting a new subscription"```?

Reply:
maybe better raise some error

Reply:
I think the expected behaviour and previous behaviour from the other libraries is that you can re-subscribe but that will override existing subscriptions (just like here).

An `IllegalStateError` is raised if `assign` is called after `subscribe` but I think this should be fine. 🤔

Reply:
mmm lets just add a log of warning in that case :)